app/routes.py

```diff
@@ -1,2 +1,225 @@
+    slack_headers = {
+        "Authorization": f"Bearer {slack_token}"
+    }
+    slack_query_params = {
+        "channel": "task-list-bot",
+        "text": f"Someone just completed the task {task.title}."
+    }
+    requests.post(slack_path, params=slack_query_params, headers=slack_headers)
+
+    return make_response(response, 200)
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def handle_incomplete_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    task.completed_at = None
+    db.session.commit()
+    response = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False
+        }
+    }
+    return make_response(response, 200)
+
+''''''''''''''''''''''''''''''''''''
+# # # # # # GOAL ROUTES # # # # # #
+''''''''''''''''''''''''''''''''''''
+
+@goals_bp.route("", methods=["POST", "GET"])
+def handle_goals():
+    if request.method == "POST":
+        request_body = request.get_json()
+        title = request_body.get("title")
+
+        if not title:
+            response_body = {"details": "Invalid data"}
+            return make_response(response_body, 400)
+
+        new_goal = Goal(title=title)
+        db.session.add(new_goal)
+        db.session.commit()
+        db_goal = {
+            "goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }
+        }
+        return db_goal, 201
+    elif request.method == "GET":
+        goals = Goal.query.all()
+
+        goals_response = []
+        for goal in goals:
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+        goals_json_response = jsonify(goals_response)
+        return make_response(goals_json_response, 200)
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        selected_goal = {
+            "goal": {
+                "id": goal.goal_id,
+                "title": goal.title}
+        }
+        return make_response(selected_goal, 200)
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+
+        goal.title = request_body["title"]
+
+        updated_goal = {
+            "goal": {
+                "id": goal.goal_id,
+                "title": goal.title
+            }
+        }
+
+        db.session.commit()
+        return make_response(updated_goal, 200)
+
+    elif request.method == "DELETE":
+        response = {
+            "details":
+            f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response(response, 200)
```

Comment:
To get Wave 6 passing start with:

-  setting up the route after this line `@goal_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])`
- make sure to implement the database relationship and goal_id in Models.py
- you've done a great job with the previous methods. Follow what you've been doing and the ReadME to get it implemented.